qiskit/circuit/library/standard_gates/i.py

```diff
@@ -51,3 +55,17 @@ def __array__(self, dtype=None):
         """Return a numpy.array for the identity gate."""
         return numpy.array([[1, 0],
                             [0, 1]], dtype=dtype)
+
+    def _define(self):
+        if not IGate._ID_IMPLICIT_DELAY_WARN_SHOWN:
+            warnings.warn('Support for implicitly converting IGates to single pulse length '
+                          'Delays has been deprecated and will be removed in a future release. '
+                          'Use the Delay instruction or QuantumCircuit.delay instead.',
+                          DeprecationWarning)
+            IGate._ID_IMPLICIT_DELAY_WARN_SHOWN = True
+
+        # pylint: disable=cyclic-import
+        from qiskit.circuit.quantumcircuit import QuantumCircuit
+        qc = QuantumCircuit(1)
+        qc.delay(320, 0, unit='dt')
```

Comment:
The standard `id` gate returned by all IBM backends is of duration `160`. Should the delay be this instead?

Reply:
Ah, interesting. I had looked at `armonk.defaults().instruction_schedule_map.get('id', 0).duration` which reports `320`. Other devices do report `160` though.